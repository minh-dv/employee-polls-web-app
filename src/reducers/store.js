import { configureStore } from "@reduxjs/toolkit";
import questionReducer from "./questionSlice";
import userReducer from "./userSlice";
import authReducer from "./authSlice";

export default configureStore({
  reducer: {
    authenticatedUser: authReducer,
    users: userReducer,
    questions: questionReducer,
  },
});
