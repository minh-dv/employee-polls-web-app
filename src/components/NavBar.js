import React from "react";
import { useDispatch } from "react-redux";
import { Link, useNavigate } from "react-router-dom";
import { logout } from "../reducers/authSlice";
import { UserInfo } from "./UserInfo";

export const NavBar = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const loggedInUser = JSON.parse(localStorage.getItem("loggedInUser"));

  const handleLogout = () => {
    dispatch(logout());
    localStorage.removeItem("loggedInUser");
    navigate("/login");
  };
  return (
    <nav className="flex justify-between border-b border-gray-400 mb-10">
      <div className="flex gap-5 ml-5">
        <Link
          to="/"
          className="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
        >
          Home
        </Link>
        <Link
          to="/leaderboard"
          className="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
        >
          Leaderboard
        </Link>
        <Link
          to="/new"
          className="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
        >
          New Poll
        </Link>
      </div>
      <div className="mr-5 flex gap-5">
        <span className="font-medium px-3 py-2 text-slate-700">
          <UserInfo user={loggedInUser} />
        </span>
        <button
          onClick={handleLogout}
          className="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
        >
          Logout
        </button>
      </div>
    </nav>
  );
};
