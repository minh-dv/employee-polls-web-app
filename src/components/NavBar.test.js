import { render } from "@testing-library/react";
import store from "../reducers/store";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import React from "react";
import { NavBar } from "./NavBar";

describe("NavBar", () => {
  it("should render the component", () => {
    const user = {
      id: "sarahedo",
      password: "password123",
      name: "Sarah Edo",
      avatarURL: "https://github.com/sarah.png",
      answers: {
        "8xf0y6ziyjabvozdd253nd": "optionOne",
        "6ni6ok3ym7mf1p33lnez": "optionOne",
        am8ehyc8byjqgar0jgpub9: "optionTwo",
        loxhs1bqm25b708cmbf3g: "optionTwo",
      },
      questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
    };

    localStorage.setItem("loggedInUser", JSON.stringify(user));

    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });

  it("should display username of logged in user", () => {
    const user = {
      id: "sarahedo",
      password: "password123",
      name: "Sarah Edo",
      avatarURL: "https://github.com/sarah.png",
      answers: {
        "8xf0y6ziyjabvozdd253nd": "optionOne",
        "6ni6ok3ym7mf1p33lnez": "optionOne",
        am8ehyc8byjqgar0jgpub9: "optionTwo",
        loxhs1bqm25b708cmbf3g: "optionTwo",
      },
      questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
    };

    localStorage.setItem("loggedInUser", JSON.stringify(user));

    const component = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    );

    const userNameDivElement = component.getByTestId("user-name");
    const userIdDivElement = component.getByTestId("user-id");
    expect(userNameDivElement.textContent).toBe("Sarah Edo");
    expect(userIdDivElement.textContent).toBe("ID: sarahedo");
  });
});
