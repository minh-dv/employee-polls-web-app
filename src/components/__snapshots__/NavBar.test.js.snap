// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="flex justify-between border-b border-gray-400 mb-10"
      >
        <div
          class="flex gap-5 ml-5"
        >
          <a
            class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
            href="/"
          >
            Home
          </a>
          <a
            class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <a
            class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
            href="/new"
          >
            New Poll
          </a>
        </div>
        <div
          class="mr-5 flex gap-5"
        >
          <span
            class="font-medium px-3 py-2 text-slate-700"
          >
            <div
              class="flex items-center"
            >
              <div
                class="flex-shrink-0 h-12 w-12"
              >
                <img
                  alt="sarahedo's avatar"
                  class="h-full w-full rounded-full"
                  src="https://github.com/sarah.png"
                />
              </div>
              <div
                class="ml-4"
              >
                <div
                  class="text-lg font-medium text-gray-900"
                  data-testid="user-name"
                >
                  Sarah Edo
                </div>
                <div
                  class="text-gray-600"
                  data-testid="user-id"
                >
                  ID: 
                  sarahedo
                </div>
              </div>
            </div>
          </span>
          <button
            class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
          >
            Logout
          </button>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="flex justify-between border-b border-gray-400 mb-10"
    >
      <div
        class="flex gap-5 ml-5"
      >
        <a
          class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
          href="/"
        >
          Home
        </a>
        <a
          class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
          href="/leaderboard"
        >
          Leaderboard
        </a>
        <a
          class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
          href="/new"
        >
          New Poll
        </a>
      </div>
      <div
        class="mr-5 flex gap-5"
      >
        <span
          class="font-medium px-3 py-2 text-slate-700"
        >
          <div
            class="flex items-center"
          >
            <div
              class="flex-shrink-0 h-12 w-12"
            >
              <img
                alt="sarahedo's avatar"
                class="h-full w-full rounded-full"
                src="https://github.com/sarah.png"
              />
            </div>
            <div
              class="ml-4"
            >
              <div
                class="text-lg font-medium text-gray-900"
                data-testid="user-name"
              >
                Sarah Edo
              </div>
              <div
                class="text-gray-600"
                data-testid="user-id"
              >
                ID: 
                sarahedo
              </div>
            </div>
          </div>
        </span>
        <button
          class="font-medium px-3 py-2 text-slate-700 rounded-lg hover:bg-slate-100 hover:text-slate-900"
        >
          Logout
        </button>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
